### PR TITLE
fix bkenv has no effect bug when start bookie using bin/pulsar-daemon

### PR DESCRIPTION
**Pull request by hangc0276**
when i start bookie using `bin/pulsar-daemon start bookie` command, i found the bookie environment variable defined in conf/bkenv.sh has no effect, but using $PULSAR_MEM and $PULSAR_GC instead.

Then i check the bin/pulsar shell script, i found as follow:
```
elif [ $COMMAND == "bookie" ]; then
    PULSAR_LOG_FILE=${PULSAR_LOG_FILE:-"bookkeeper.log"}
    # Pass BOOKIE_EXTRA_OPTS option defined in pulsar_env.sh
    OPTS="$OPTS $BOOKIE_EXTRA_OPTS"
    exec $JAVA $OPTS -Dpulsar.log.file=$PULSAR_LOG_FILE org.apache.bookkeeper.proto.BookieServer --conf $PULSAR_BOOKKEEPER_CONF $@
```
but in pulsar_env.sh, the `$BOOKIE_EXTRA_OPTS` defined as follow:
```
# Set BOOKIE_EXTRA_OPTS option here to ensure that all pulsar scripts can work seamless with bookkeeper

# Extra options to be passed to the jvm
BOOKIE_EXTRA_OPTS="${BOOKIE_EXTRA_OPTS} -Dio.netty.leakDetectionLevel=disabled -Dio.netty.recycler.maxCapacity.default=1000 -Dio.netty.recycler.linkCapacity=1024"
```
`$BOOKIE_EXTRA_OPTS` do not define `$BOOKIE_MEM` and `$BOOKIE_GC`, then the pulsar script will using `$PULSAR_MEM` and `$PULSAR_GC` instead to start bookkeeper.

So, i suggest to load conf/bkenv.sh in bin/pulsar script and using `$BOOKIE_EXTRA_OPTS` in conf/bkenv.sh instread of `$BOOKIE_EXTRA_OPTS` in bin/pulsar_env.sh

### CODE REVIEW

File: bin/pulsar [hunk 1]
```diff
@@ -44,12 +44,18 @@ FUNCTIONS_EXTRA_DEPS_DIR=${PULSAR_FUNCTIONS_EXTRA_DEPS_DIR:-"${DEFAULT_FUNCTIONS
...
 PRESTO_HOME=${PULSAR_HOME}/lib/presto
 
-
+# Check pulsar env and load pulser_env.sh
 if [ -f "$PULSAR_HOME/conf/pulsar_env.sh" ]
 then
     . "$PULSAR_HOME/conf/pulsar_env.sh"
 fi
 
+# Check bookkeeper env and load bkenv.sh
```
Comment (sijie) [inline bin/pulsar]: Move this before line 47 for backward compatibility consideration.
So if people already define environments in pulsar_env, those env vars will not be overridden by bkenv.sh.

File: bin/pulsar-perf [hunk 2]
```diff
@@ -29,6 +29,12 @@ then
     . "$PULSAR_HOME/conf/pulsar_env.sh"
 fi
 
+# Check bookkeeper env and load bkenv.sh
```
Comment (sijie) [inline bin/pulsar-perf]: same comment as in bin/pulsar

File: bin/pulsar-daemon [hunk 3]
```diff
@@ -45,6 +45,12 @@ then
     . "$PULSAR_HOME/conf/pulsar_env.sh"
 fi
 
+# Check bookkeeper env and load bkenv.sh
```
Comment (sijie) [inline bin/pulsar-daemon]: same comment as above

File: src/pulsar-io-gen [hunk 4]
```diff
@@ -28,6 +28,12 @@ then
     . "$PULSAR_HOME/conf/pulsar_env.sh"
 fi
 
+# Check bookkeeper env and load bkenv.sh
```
Comment (sijie) [inline src/pulsar-io-gen]: same comment as above

**Comment by merlimat** [comment]
This change is causing several side effects and I think it should be reverted.

**Comment by roman-popenov** [comment]
@merlimat what are the side effects?

**Pull request merged**